libsolutil/CommonData.h

```diff
@@ -265,6 +265,39 @@ decltype(auto) valueOrDefault(MapType&& _map, KeyType const& _key, ValueType&& _
 	return (it == _map.end()) ? _defaultValue : it->second;
 }
 
+namespace detail
+{
+template<typename Callable>
+struct MapTuple
+{
+	Callable callable;
+	template<typename TupleType>
+	decltype(auto) operator()(TupleType&& _tuple) {
+		using PlainTupleType = std::remove_cv_t<std::remove_reference_t<TupleType>>;
+		return operator()(
+			std::forward<TupleType>(_tuple),
+			std::make_index_sequence<std::tuple_size_v<PlainTupleType>>{}
+		);
+	}
+private:
+	template<typename TupleType, size_t... I>
+	decltype(auto) operator()(TupleType&& _tuple, std::index_sequence<I...>)
+	{
+		return callable(std::get<I>(std::forward<TupleType>(_tuple))...);
+	}
+};
+}
+
+/// Wraps @a _callable, which takes multiple arguments, into a callable that takes a single tuple of arguments.
+/// Since structured binding in lambdas is not allowed, i.e. [](auto&& [key, value]) { ... } is invalid, this allows
+/// to instead use mapTuple([](auto&& key, auto&& value) { ... }).
```

Comment:
So yeah, @chriseth: Weren't you actually complaining about ``[](auto&& [key, value]) { .. }`` not being possible recently :-)? This is a workaround for that :-)...

And yeah - look at the latest commit here, that's this in use - it actually helps in more cases than in that ``erase_if``...

But if you think this is too much and too complicated or still not maximally readable, just let me know, then I'll just drop that last commit and add an ``eraseValue`` helper instead, I really don't mind too much at this point :-).